pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -226,6 +323,68 @@ func (s *GameServerSelector) Validate(field string) ([]metav1.StatusCause, bool)
 		}
 	}
 
+	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) && (s.Counters != nil || s.Lists != nil) {
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: "Feature CountsAndLists must be enabled if Counters or Lists are specified",
+			Field:   field,
+		})
+	}
+
+	if runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
+		if s.Counters != nil {
+			for _, counterSelector := range s.Counters {
+				if counterSelector.MinCount < 0 {
+					causes = append(causes, metav1.StatusCause{
+						Type:    metav1.CauseTypeFieldValueInvalid,
+						Message: "CounterSelector.MinCount must be greater than zero",
+						Field:   field,
+					})
+				}
+				if counterSelector.MaxCount < 0 {
+					causes = append(causes, metav1.StatusCause{
+						Type:    metav1.CauseTypeFieldValueInvalid,
+						Message: "CounterSelector.MaxCount must be greater than zero",
+						Field:   field,
+					})
+				}
+				if counterSelector.MinAvailable < 0 {
+					causes = append(causes, metav1.StatusCause{
+						Type:    metav1.CauseTypeFieldValueInvalid,
+						Message: "CounterSelector.MinAvailable must be greater than zero",
+						Field:   field,
+					})
+				}
+				if counterSelector.MaxAvailable < 0 {
+					causes = append(causes, metav1.StatusCause{
+						Type:    metav1.CauseTypeFieldValueInvalid,
+						Message: "CounterSelector.MaxAvailable must be greater than zero",
+						Field:   field,
+					})
```

Comment:
Do we need to validate `Min* <= Max*`?

Reply:
Sure, I can add that check in.

Reply:
I'm now running into lint error on build "pkg/apis/allocation/v1/gameserverallocation.go:282:1: cyclomatic complexity 31 of func `(*GameServerSelector).Validate` is high (> 30) (gocyclo)". Any suggestions for reducing cyclomatic complexity besides taking out checks? Do you see any disadvantage to changing CounterSelector and ListSelector fields to uint so that I don't need to do any non-negative checks?

Reply:
So looks like the check is on method level? I'm wondering whether having a helper func that takes a single `counterSelector` to do the validation (line 341 to 382) and calling this helper func in the for loop (line 340) would help.